components/server/src/auth/jwt.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import * as jsonwebtoken from "jsonwebtoken";
+import { Config } from "../config";
+import { inject, injectable } from "inversify";
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+
+const algorithm: jsonwebtoken.Algorithm = "RS512";
+
+@injectable()
+export class AuthJWT {
+    @inject(Config) protected config: Config;
+
+    async sign(subject: string, payload: object | Buffer, expiresIn: string = `${24 * 7}h`): Promise<string> {
+        const opts: jsonwebtoken.SignOptions = {
+            algorithm,
+            expiresIn,
+            issuer: this.config.hostUrl.url.hostname,
```

Comment:
Let's use the full URL of the issuer, not just the hostname here, i.e. the scheme is missing to be aligned with other other libraries and discovery in general (there the ".well-known" part is basically appended).

While the [spec](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1) is talking about StringOrURI, usually it's the URI you are interested in when it comes to discovery and such.

Reply:
Fixed